examples/pytorch/model_zoo/utils.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import torch
+from sklearn.metrics import roc_auc_score
+import dgl
+
+class Meter():
+    def __init__(self, mode=None):
+        self.mode = mode
+        self.mask = []
+        self.y_pred = []
+        self.y_true = []
+
+    def update(self, y_pred, y_true, mask):
+        self.y_pred.append(y_pred)
+        self.y_true.append(y_true)
+        self.mask.append(mask)
+
+    def roc_auc_averaged_over_tasks(self):
+
+        mask = torch.cat(self.mask, dim=0)
+        y_pred = torch.cat(self.y_pred, dim=0)
+        y_true = torch.cat(self.y_true, dim=0)
+        y_pred = torch.sigmoid(y_pred)
+        n_tasks = y_true.shape[1]
+        total_score = 0
+        for task in range(n_tasks):
+            task_w = mask[:, task]
+            task_y_true = y_true[:, task][task_w != 0].cpu().numpy()
+            task_y_pred = y_pred[:, task][task_w != 0].cpu().detach().numpy()
+            total_score += roc_auc_score(task_y_true, task_y_pred)
+        return total_score / n_tasks
+
+
+class EarlyStopping:
+    def __init__(self, patience=10, filename="es_checkpoint.pt"):
```

Comment:
To be consistent, I'd suggest changing ``es_checkpoint.pt`` to ``es_checkpoint.pth`` unless you want to change the pre-trained file.

Reply:
Done